### PR TITLE
Nisarg - SL button added

### PR DESCRIPTION
**Pull request by nisarg-solutelabs**

### CODE REVIEW

File: lib/components/button/sl_button.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+              ),
+            ),
+            child: Text(
+              text,
+              style: textStyle,
+            ),
+          ),
+        ),
+      );
+}
```
Comment (snehasen-solutelabs) [inline lib/components/button/sl_button.dart:76]: add new line at the end or format this file.

File: lib/components/button/sl_button.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  final Color backgroundColor;
+
+  /// textstyle of the button text
+  final TextStyle? textStyle;
+
+  /// border radius of the button
+  final double borderRadius;
+
+  ///height of the button
+  final double height;
```
Comment (snehasen-solutelabs) [inline lib/components/button/sl_button.dart:38]: add width variable too.

File: lib/components/button/sl_button.dart [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                RoundedRectangleBorder(
+                  borderRadius: BorderRadius.circular(
+                    borderRadius,
+                  ),
+                ),
+              ),
+              backgroundColor: MaterialStateProperty.all(backgroundColor),
+              side: MaterialStateProperty.all(
+                BorderSide(
+                  color: borderColor,
```
Comment (snehasen-solutelabs) [inline lib/components/button/sl_button.dart:65]: if border colour is null then ? and similar to other variable , add null case.

**Review by snehasen-solutelabs — CHANGES_REQUESTED** [review]

File: lib/components/button/sl_button.dart (hunk 3 above)
Comment (nisarg-solutelabs) [reply lib/components/button/sl_button.dart:65]: I have added default values in parameters.
const SlButton({
    required this.text,
    required this.onTap,
    this.textStyle,
    this.borderColor = SLColors.white,
    this.backgroundColor = SLColors.white,
    this.borderRadius = 6,
    this.height = 42,
    this.buttonPadding,
    Key? key,
  }) : super(
          key: key,
        );

**Pull request merged**